src/gui/GUIManager.ts

```diff
@@ -0,0 +1,676 @@
+import { htmlToText } from "html-to-text";
+import * as htmlEscape from "escape-html";
+
+import { JSDOM } from "jsdom";
+import { LogService } from "matrix-bot-sdk";
+import { Mjolnir } from "../Mjolnir";
+import { limitLength } from "../utils";
+
+/// Regexp, used to extract the action label from an action reaction
+/// such as `⚽ Kick user @foobar:localhost from room [kick-user]`.
+const REACTION_ACTION = /\[([a-z-]*)\]$/;
```

Comment:
Could we look at https://github.com/matrix-org/mjolnir/pull/137#discussion_r732246912 again? seems like it's been merged into spec now

Reply:
Good point. That would definitely be nicer.

This means that the feature will only work with recent versions of Synapse, though.